src/app/qgsappwindowmanager.h

```diff
@@ -0,0 +1,62 @@
+/***************************************************************************
+                             qgswindowmanagerinterface.h
+                             ---------------------------
+    Date                 : September 2018
+    Copyright            : (C) 2018 Nyall Dawson
+    Email                : nyall dot dawson at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSAPPWINDOWMANAGER_H
+#define QGSAPPWINDOWMANAGER_H
+
+#include "qgis.h"
+#include "qgswindowmanagerinterface.h"
+#include <QPointer>
+
+class QgsStyleManagerDialog;
+class QgsLayoutManagerDialog;
+
+/**
+ * \ingroup gui
+ * \brief Implementation of QgsWindowManagerInterface for the QGIS application.
+ */
+class QgsAppWindowManager : public QgsWindowManagerInterface
+{
+  public:
+
+    //! Application-only QGIS dialogs
+    enum ApplicationDialog
+    {
+      DialogLayoutManager = 0, //!< Layout manager dialog
+    };
```

Comment:
So the "application dialogs" are dialogs that are not supposed to be requested by gui library?

Reply:
That's correct - it's separating out app only dialogs from those in the GUI library